torch_xla/csrc/aten_autograd_ops.cpp

```diff
@@ -18,6 +18,42 @@ bool IsNonTrivialDilation(at::IntArrayRef dilation) {
 
 namespace aten_autograd_ops {
 
+torch::Tensor EinsumAutogradFunction::forward(
+    torch::autograd::AutogradContext* ctx, const c10::string_view equation,
+    at::TensorList tensors) {
+  ctx->saved_data["equation"] = equation;
+
+  torch::autograd::variable_list vars;
+  for (auto const& tensor : tensors) {
+    vars.push_back(tensor);
+  }
+  ctx->save_for_backward(vars);
+
+  auto xla_tensors = bridge::GetXlaTensors(absl::MakeSpan(tensors));
+  auto output = XLATensor::einsum(equation, xla_tensors);
+  return bridge::AtenFromXlaTensor(output);
+}
+
+torch::autograd::variable_list EinsumAutogradFunction::backward(
+    torch::autograd::AutogradContext* ctx,
+    torch::autograd::variable_list grad_output) {
+  auto equation = ctx->saved_data["equation"].toString()->string_view();
+  auto tensors = ctx->get_saved_variables();
+  auto xla_tensors = bridge::GetXlaTensors(absl::MakeSpan(tensors));
+
+  auto outputs = XLATensor::einsum_backward(
+      bridge::GetXlaTensor(grad_output[0]), xla_tensors, equation);
+
+  torch::Tensor undef;
+  torch::autograd::variable_list grad_inputs = {
+      undef, bridge::AtenFromXlaTensor(std::get<0>(outputs))};
+  if (std::get<1>(outputs).defined()) {
```

Comment:
can you add a comment here to explain this check?

Reply:
Done